packages/webview_flutter/webview_flutter_wkwebview/lib/src/webkit_webview_controller.dart

```diff
@@ -630,33 +630,65 @@ class WebKitWebViewController extends PlatformWebViewController {
   }
 
   Future<void> _injectConsoleOverride() {
+    // Within overrideScript, a series of console output methods such as
+    // console.log will be rewritten to pass the output content to the Flutter
+    // end.
+    //
+    // These output contents will first be serialized through JSON.stringify(),
+    // but if the output content contains cyclic objects, it will encounter the
+    // following error.
+    // TypeError: JSON.stringify cannot serialize cyclic structures.
+    // See https://github.com/flutter/flutter/issues/144535.
+    //
+    // Considering this is just looking at the logs printed via console.log,
+    // the cyclic object is not important, so remove it.
+    // Therefore, the replacer parameter of JSON.stringify() is used and the
+    // removeCyclicObject method is passed in to solve the error.
     const WKUserScript overrideScript = WKUserScript(
       '''
-function log(type, args) {
-  var message =  Object.values(args)
-      .map(v => typeof(v) === "undefined" ? "undefined" : typeof(v) === "object" ? JSON.stringify(v) : v.toString())
-      .map(v => v.substring(0, 3000)) // Limit msg to 3000 chars
-      .join(", ");
-
-  var log = {
-    level: type,
-    message: message
-  };
-
-  window.webkit.messageHandlers.fltConsoleMessage.postMessage(JSON.stringify(log));
-}
+var _flutter_webview_plugin_overrides = _flutter_webview_plugin_overrides || {
+  removeCyclicObject: function() {
+    const levelObjects = [];
```

Comment:
Based on the description in the PR comment, `traversalStack` would probably be a clearer name.

Reply:
Thx.